libs/s25main/mapGenerator/Harbors.cpp

```diff
@@ -86,19 +78,29 @@ namespace rttr { namespace mapGenerator {
             {
                 if(isCoast(pt))
                 {
-                    auto coast = Collect(map.textures, pt, isCoast);
+                    const auto coast = Collect(map.textures, pt, isCoast);
                     coasts.push_back(coast);
                     visited.insert(coast.begin(), coast.end());
                 }
             }
         }
+
+        const auto distanceToRiver = DistancesTo(join(rivers), map.size);
+        const auto closeToRiver = [&distanceToRiver](const MapPoint& pt) { return distanceToRiver[pt] < 5; };
+
+        for(std::vector<MapPoint>& coast : coasts)
+        {
+            coast.erase(std::remove_if(coast.begin(), coast.end(), closeToRiver), coast.end());
+        }
+
         return coasts;
     }
 
     void PlaceHarbors(Map& map, const std::vector<River>& rivers, int coastSize, int nodesPerHarbor)
     {
-        auto coasts = FindCoastlines(map, rivers);
-        for(auto coast : coasts)
+        const auto coasts = FindCoastlines(map, rivers);
+        std::vector<MapPoint> harbors;
+        for(const auto& coast : coasts)
```

Comment:
Can you include the type (instead of auto) either here or in line 101? It's hard to follow otherwise, thanks! :)

Reply:
probably better in 101

Reply:
I wouldn't mind. They are close and the only important thing to know here is that `coasts` is a collection, the exact and bulky type isn't important, but for `coast` it is (more) important and the type is shorter.

Reply:
sure, makes sense